backend/controllers/investment.controller.js

```diff
@@ -0,0 +1,173 @@
+import mongoose from "mongoose";
+import Investment from '../models/investment.model.js';
+import { createTransaction } from '../middlewares/createTransactionHelper.js';
+import {errorHandler} from '../middlewares/error.js'; // Adjust the import according to your project structure
+
+// Get all investments
+export const getInvestments = async (req, res, next) => {
+    try {
+        const {page = 1, limit = 25} = req.query;
+        const investments = await Investment.find().limit(parseInt(limit)).skip((page - 1) * limit);
+        res.status(200).json({
+            success: true,
+            data: investments
+        });
+    } catch (err) {
+        next(err);
+    }
+};
+
+// Get a single investment by ID
+export const getInvestmentsByAccountNumber = async (req, res, next) => {
+    const { accountNumber } = req.params;
+
+    try {
+        const investment = await Investment.find({accountNumber});
+
+        if (!investment) {
+            return next(errorHandler(404, 'Investment not found'));
+        }
+
+        res.status(200).json({
+            success: true,
+            data: investment
+        });
+    } catch (err) {
+        next(err);
+    }
+};
+
+// Create a new investment
+export const createInvestment = async (req, res, next) => {
+    const investment = req.body;
+    const session = await mongoose.startSession();
+
+    try {
+        session.startTransaction();
+
+        // Create and save the investment
+        const newInvestment = new Investment(investment);
+        const savedInvestment = await newInvestment.save({ session });
+
+        // Create the transaction
+        const transactionData = { amount: investment.amount, type: 'investment', accountNumber: investment.accountNumber, cardId : investment.cardId, invoice: "Invested Certain amount", description: investment.description };
+        await createTransaction(transactionData);
+
+        await session.commitTransaction();
+        session.endSession();
+
+        res.status(201).json({
+            success: true,
+            data: { Investment: savedInvestment }
+        });
+    } catch (err) {
+        await session.abortTransaction();
+        session.endSession();
+        next(err);
+    }
+};
+
+
+// Update an existing investment
+export const updateInvestment = async (req, res, next) => {
+    const { accountNumber } = req.params;
+    const updateFields = req.body;
+
+    try {
+        const updatedInvestment = await Investment.findOneAndUpdate(
+            { accountNumber},
+            { $set: updateFields },
+            { new: true, runValidators: true }
```

Comment:
The updateInvestment data gets updated correctly, but both the previous and updated data are displayed. The data should overwrite the previous information to display a single format.